src/domain/models/Entity.ts

```diff
@@ -0,0 +1,136 @@
+import objectHash from 'object-hash';
+import EntityDiscardedError from '../exceptions/EntityDiscardedError';
+import Props from '../../interfaces/Props';
+
+/**
+ * Represent the base element in the domain model, for entities and its aggregates.
+ */
+export default abstract class Entity<Id_T> {
+  private _id: Id_T;
```

Comment:
Is it necessary to use `_` before the attr even being declared as `private`? Sorry, do not know much about `.ts`

Reply:
Not really, but as I'm using the get/set properties (like python does), I need to distinguish the field itself from the property in this case. The best practice is to use an underscore prefix.

Reply:
Ok, i got it